x-pack/plugins/observability_solution/apm/public/components/shared/explore_logs_button/logs_explorer_locator_config.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AllDatasetsLocatorParams } from '@kbn/deeplinks-observability';
+import { LocatorPublic } from '@kbn/share-plugin/common';
+import { getRouterLinkProps } from '@kbn/router-utils';
+import { RouterLinkProps } from '@kbn/router-utils/src/get_router_link_props';
+
+export const buildLogsExplorerLocatorConfig = ({
+  locator,
+  kuery,
+  from,
+  to,
+}: {
+  locator: LocatorPublic<AllDatasetsLocatorParams>;
+  kuery?: string;
+  from: string;
+  to: string;
+}): {
+  logsExplorerLinkProps: RouterLinkProps;
+} => {
+  const params: AllDatasetsLocatorParams = {
+    timeRange: {
+      from,
+      to,
+    },
+    ...(kuery && {
+      query: { language: 'kuery', query: kuery },
+    }),
+  };
+
+  const urlToLogsExplorer = locator.getRedirectUrl(params);
+
+  const navigateToLogsExplorer = () => {
+    locator.navigate(params) as Promise<void>;
+  };
+
+  const logsExplorerLinkProps = getRouterLinkProps({
+    href: urlToLogsExplorer,
+    onClick: navigateToLogsExplorer,
```

Comment:
Q: Apparently, we're not using `onClick` yet. Should we keep it or do we already have plans to use it in the future?

Reply:
AFAIK, no plans atm. I thought we might need telemetry but I can remove it and added back if needed it.

Reply:
getRouterLinkProps requires the onClick
 
 ```
  Property 'onClick' is missing in type '{ href: string; }' but required in type 'GetRouterLinkPropsDeps'
```